### ISSUE TITLE
Suggestions for Music dataset

### ISSUE THREAD

**Issue by zdchu** [seunghyukcho/CoNAL-pytorch#1]
Hi there,

Thanks for your interest in our paper! The poor performance on the Music dataset is mainly caused by the classifier. You can try to add two Batch Norm before the two linear layers.

**Comment by seunghyukcho** [comment]
Thanks for advices!

Do you mean adding two Batch Norm in Music classifier is this?

original code:
```python
self.layers = nn.Sequential(
            nn.Linear(124, self.n_units),
            nn.BatchNorm1d(self.n_units, affine=False),
            nn.ReLU(),
            nn.Dropout(self.dropout),
            nn.Linear(self.n_units, self.n_class),
            nn.Softmax(dim=-1)
        )
```

fixed code:
```python
self.layers = nn.Sequential(
            nn.BatchNorm1d(124, affine=False),  # added!
            nn.Linear(124, self.n_units),
            nn.BatchNorm1d(self.n_units, affine=False),
            nn.ReLU(),
            nn.Dropout(self.dropout),
            nn.Linear(self.n_units, self.n_class),
            nn.Softmax(dim=-1)
        )
```

**Comment by zdchu** [comment]
I added the first batch norm before the first linear layer and it is helpful.

On Feb 3, 2021, at 3:39 AM, 조승혁 <notifications@github.com> wrote:

﻿

Thanks for advices!

Do you mean adding two Batch Norm in Music classifier is this?

original code:

self.layers = nn.Sequential(
            nn.Linear(124, self.n_units),
            nn.BatchNorm1d(self.n_units, affine=False),
            nn.ReLU(),
            nn.Dropout(self.dropout),
            nn.Linear(self.n_units, self.n_class),
            nn.Softmax(dim=-1)
        )

fixed code:

self.layers = nn.Sequential(
            nn.BatchNorm1d(124, affine=False),  # added!
            nn.Linear(124, self.n_units),
            nn.BatchNorm1d(self.n_units, affine=False),
            nn.ReLU(),
            nn.Dropout(self.dropout),
            nn.Linear(self.n_units, self.n_class),
            nn.Softmax(dim=-1)
        )

**Comment by zdchu** [comment]
Yes! But I add the second BN after the ReLU and dropout. You can give it a try!

**Comment by seunghyukcho** [comment]
Thanks for more details. However, I've followed your advices but get accuracy near to `0.70`. So I have some question for more details about your implementation.

-  Did you set the `affine` parameter `False` for all batch normalization?
-  Did you made any weight initialization?
-  How was your classifier accuracy on train set? In my case, it was near `0.80` even below your experiments result
-  Is there any batch normalization in auxiliary network and noise adaptation layers?
-  How did you split the train / valid / test set in Music? I'm using test set for validation.
-  Did you used the 124 feature music data in the link, or the original 30 second music data?

**Comment by zdchu** [comment]
Yes, I set affine=False in BN. I got near 0.81 on the training set. The other settings are the same.

**Comment by seunghyukcho** [comment]
Now I achieved average accuracy `0.75` with similar train accuracy with yours. I'll try more techniques like weight initialization.

One more question is that did you used batch size 1024, learning rate 1e-2 for Music?

And did your Music training set also had 700 train data?

**Comment by lazyloafer** [comment]
Hi Chu,
I'm reading your paper (**Improve Learning from Crowds via Generative Augmentation in KDD'21**). I want to find out whether the structure of the generator G in Fig. 1 is similar to that in CoNAL? In other words, how does G combine instance diffusion, annotator expertise, true labels of the instances, and random noise? @zdchu
